src/Criteo.OpenApi.Comparator/Comparators/OperationComparator.cs

```diff
@@ -257,30 +257,75 @@ private static int FindParameterIndex(OpenApiParameter parameter, IList<OpenApiP
         /// <summary>
         /// Finds parameter name in the list of operation parameters or global parameters
         /// </summary>
-        /// <param name="name">name of the parameter to search</param>
+        /// <param name="key">The parameter used as a key. It is expected that this parameter comes from the opposite
+        /// file. The parameter that has the same name and is most similar to this parameter will be returned</param>
         /// <param name="operationParameters">list of operation parameters to search</param>
         /// <param name="documentParameters">Dictionary of global parameters to search</param>
         /// <returns>Swagger Parameter if found; otherwise null</returns>
         private static OpenApiParameter FindParameter(
-            string name,
+            OpenApiParameter key,
             IEnumerable<OpenApiParameter> operationParameters,
             IDictionary<string, OpenApiParameter> documentParameters)
         {
+            string name = key.Name;
             if (name == null || operationParameters == null)
                 return null;
 
+            var candidateParameters = new List<OpenApiParameter>();
             foreach (var parameter in operationParameters)
             {
                 if (name.Equals(parameter.Name))
-                    return parameter;
+                    candidateParameters.Add(parameter);
+                else
+                {
+                    var referencedParameter = parameter.Reference.Resolve(documentParameters);
 
-                var referencedParameter = parameter.Reference.Resolve(documentParameters);
+                    if (referencedParameter != null && name.Equals(referencedParameter.Name))
+                        candidateParameters.Add(referencedParameter);
+                }
+            }
 
-                if (referencedParameter != null && name.Equals(referencedParameter.Name))
-                    return referencedParameter;
+            if (!candidateParameters.Any())
+                return null;
+
+            if(candidateParameters.Count == 0)
+                return candidateParameters[0];
```

Comment:
That will throw an exception. I think you meant:
```
if(candidateParameters.Count == 1)
  return candidateParameters[0];
```

Reply:
Good catch, I'll add a unity test that causes this failure then fix it.

Reply:
Just realized that 

```
if(candidateParameters.Count == 1)
  return candidateParameters[0];
```

Never actually happened because of the Any check above. So it was skipped and `GetClosestFunctionally` was just returning the 1 candidate any way.